docs/src/selenium-grid.md

```diff
@@ -3,66 +3,174 @@ id: selenium-grid
+:::note
+Before connecting Playwright to your Selenium Grid, make sure that grid works with [Selenium WebDriver](https://www.selenium.dev/documentation/webdriver/). For example, run [one of the examples](https://github.com/SeleniumHQ/selenium/tree/trunk/javascript/node/selenium-webdriver/example) and pass `SELENIUM_REMOTE_URL` environment variable.
 
-```bash js
-# Playwright Test
-SELENIUM_REMOTE_URL=http://internal.grid:4444/wd/hub npx playwright test
+If webdriver example does not work, look for any errors at your Selenium hub/node/standalone output and search [Selenium issues](https://github.com/SeleniumHQ/selenium/issues) for a possible solution.
+:::
+
+## Starting Selenium Grid
 
-# Playwright Library
-SELENIUM_REMOTE_URL=http://internal.grid:4444/wd/hub node script.js
+If you run distributed Selenium Grid, Playwright needs selenium nodes to be registered with an accessible address, so that it could connect to the browsers. To make sure it works as expect, set `SE_NODE_GRID_URL` environment variable pointing to the hub when running selenium nodes.
+
+```bash
+# Start selenium node
+SE_NODE_GRID_URL="http://<selenium-hub-ip>:4444/wd/hub" java -jar selenium-server-<version>.jar node
 ```
 
-```bash python
-# With Pytest
-SELENIUM_REMOTE_URL=http://internal.grid:4444/wd/hub pytest --browser chromium
+## Connecting Playwright to Selenium Grid
+
+To connect Playwright to **Selenium Grid 4**, set `SELENIUM_REMOTE_URL` environment variable pointing to your Selenium Grid Hub. Note this only works for Google Chrome and Microsoft Edge.
+
+```bash js
+SELENIUM_REMOTE_URL=http://<selenium-hub-ip>:4444/wd/hub npx playwright test
+```
 
-# Plain Python
-SELENIUM_REMOTE_URL=http://internal.grid:4444/wd/hub python script.py
+```bash python
+SELENIUM_REMOTE_URL=http://<selenium-hub-ip>:4444/wd/hub pytest --browser chromium
 ```
 
 ```bash java
-SELENIUM_REMOTE_URL=http://internal.grid:4444/wd/hub mvn test
+SELENIUM_REMOTE_URL=http://<selenium-hub-ip>:4444/wd/hub mvn test
 ```
 
 ```bash csharp
-SELENIUM_REMOTE_URL=http://internal.grid:4444/wd/hub dotnet test
+SELENIUM_REMOTE_URL=http://<selenium-hub-ip>:4444/wd/hub dotnet test
 ```
 
-You don't have to change your code, just use [`method: BrowserType.launch`] as usual.
+You don't have to change your code, just use your testing harness or [`method: BrowserType.launch`] as usual.
 
 When using Selenium Grid Hub, you can [skip browser downloads](./browsers.md#skip-browser-downloads).
 
-If your grid requires additional capabilities to be set (for example, you use an external service), you can use `SELENIUM_REMOTE_CAPABILITIES` environment variable to provide JSON-serialized capabilities.
+### Passing additional capabilities
+
+If your grid requires additional capabilities to be set (for example, you use an external service), you can set `SELENIUM_REMOTE_CAPABILITIES` environment variable to provide JSON-serialized capabilities.
```

Comment:
nit: maybe link SELENIUM_REMOTE_CAPABILITIES to a selenium doc?

Reply:
Thanks

Reply:
It is a playwright thing, does not exist in selenium webdriver.